### PR TITLE
Add pkgGpgCheck callback

### PR DESCRIPTION
**Pull request by imobach**
I will merge this change when a solution for https://bugzilla.suse.com/show_bug.cgi?id=948608 is ready. But, by now, I need some feedback.

### CODE REVIEW

File: src/Callbacks_Register.cc [hunk 1]
```diff
@@ -324,6 +324,17 @@ YCPValue PkgFunctions::CallbackFinishDeltaApply( const YCPValue& args)
 }
 
 /**
+ * @builtin CallbackPkgGpgCheck
+ * @short Register callback function
+ * @param string args Name of the callback handler function. Required callback prototype is <code>void()</code>. The callback function is evaluated when the delta download has been applied.
```
Comment (lslezak) [inline src/Callbacks_Register.cc]: Just update the comment for this callback.

File: src/Callbacks.cc [hunk 2]
```diff
@@ -705,6 +707,39 @@ namespace ZyppRecipients {
...
+        // Package
+        const zypp::Package::constPtr & package_r = userData_r.get<zypp::Package::constPtr>("Package");
+        callback.addStr(userData_r.get<std::string>("Package", package_r->name()));
+
+        // Localpath
+        zypp::Pathname localpath = userData_r.get<zypp::Pathname>("Localpath");
+        callback.addStr(localpath.asString());
+
+        // Result
+        callback.addInt(userData_r.get<RpmDb::CheckPackageResult>("CheckPackageResult"));
```
Comment (lslezak) [inline src/Callbacks.cc]: I'm thinking about the callback prototype, currently the registered function has signature `foo(String, String, Fixnum)`. However, the libzypp parameter is `zypp::UserData` which is basically a Hash. So maybe we should use `foo(Hash)` instead. (There is [CB::addMap() method](https://github.com/yast/yast-pkg-bindings/blob/084a5600058c4fe96a0995e42182cc9cc3053e50/src/Callbacks.YCP.h#L203).)

If libzypp later adds some more user data we can easily pass it to YaST (simply add to a Hash) without breaking the API...

**Comment by lslezak** [comment]
LGTM

**Comment by imobach** [comment]
@lslezak Thanks for your help and review :)

**Comment by lslezak** [comment]
LGTM, thanks!

**Comment by imobach** [comment]
@lslezak Thanks for review!

**Pull request merged**